diff --git a/src/slam_toolbox_common.cpp b/src/slam_toolbox_common.cpp
index 0210b2c..b372e22 100644
--- a/src/slam_toolbox_common.cpp
+++ b/src/slam_toolbox_common.cpp
@@ -123,7 +123,7 @@ CallbackReturn SlamToolbox::on_configure(const rclcpp_lifecycle::State &)
     get_node_base_interface(),
     get_node_timers_interface());
   tf_->setCreateTimerInterface(timer_interface);
-  tfL_ = std::make_unique<tf2_ros::TransformListener>(*tf_);
+  tfL_ = std::make_unique<tf2_ros::TransformListener>(*tf_, shared_from_this());
   tfB_ = std::make_unique<tf2_ros::TransformBroadcaster>(shared_from_this());
   laser_assistant_ = std::make_unique<laser_utils::LaserAssistant>(
     shared_from_this(), tf_.get(), base_frame_);
